### DISCUSSION TITLE
how to pick objects which are outside the screen

### DISCUSSION THREAD

**Discussion by yanshjie** [visgl/deck.gl#7268] [Q&A] [answered]
Hello, dear developers. The picking APIs like ```pickObject({x, y, radius, layerIds})``` in Deck class receive pixel position parameter, which means I can only get picking info in the screen. 
But my application needs to automatically get some picking infomation which the picked position (coordinates in [lng, lat]) maybe on the screen or maybe outside the screen. 
The official doc says I can use multiple views together with multiple viewstates in one deck class, so my ideas is to create a _virtual_ view and viewstate besides the main view so that the offscreen point becomes 'onscreen' in the virtual view. 
Is that possible? I don't know how to hide the virtual view.
I an very gratitude for  your reply.

**Comment by Pessimistress** [comment, marked as answer]
You should not use the built-in picking for things not drawn to screen. The feature is raster-based and only makes sense when you are interacting with a pixel.

Look at [turf.js](http://turfjs.org) for proper geospatial tests.
